core/src/main/java/org/apache/accumulo/core/iterators/YieldCallback.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.iterators;
+
+/**
+ * This callback handles the state of yielding within an iterator
+ */
+public class YieldCallback<K> {
```

Comment:
@ivakegg for the case of a transforming iterator that supports yielding, are you thinking it would it work like the following?

```java
class MyIter implements SKVI {
  YieldCallback yc;

  void next() {
     source.next();
     if(yc.hasYielded()) {
       Key tmpKey = yc.getPositionAndReset();
       tmpKey = transform(tmpKey);
       yc.yield(tmpKey);
     }
  }
}
```